tests/testlib/s2n_connection_test_utils.c

```diff
@@ -58,7 +58,7 @@ static int buffer_read(void *io_context, uint8_t *buf, uint32_t len)
         return -1;
     }
 
-    s2n_stuffer_read_bytes(in_buf, buf, n_read);
+    POSIX_GUARD(s2n_stuffer_read_bytes(in_buf, buf, n_read));
```

Comment:
Is this correct? You'd be returning without setting errno.

Reply:
Hmm, maybe I'm not understanding one of the expected behaviors here. If this `POSIX_GUARD` is triggered, won't the errno bubble up?

Or are you suggesting that we should set this to some buffer reading specific errno. If so perhaps `EIO /* I/O error */` would be appropriate?

Reply:
If you don't set errno, there's nothing to bubble up :) We're basically implementing a syscall here, so we're responsible for errno the way an s2n library method is responsible for s2n_errno.

EIO sounds right to me.

Reply:
I often just set errno to a generic error like EIO at the beginning of the method so that we can use GUARDs, but that wasn't done here.

Reply:
> I often just set errno to a generic error like EIO at the beginning of the method so that we can use GUARDs

Wouldn't that get overridden by the `errno` stuff internal to these methods? For example
```
- stuffer_read_bytes
-